src/main/java/org/opensearch/security/state/SecurityClusterStateListeners.java

```diff
@@ -0,0 +1,128 @@
+import org.opensearch.security.configuration.ConfigurationRepository;
+import org.opensearch.threadpool.ThreadPool;
+
+public final class SecurityClusterStateListeners {
+
+    private static final Logger LOGGER = LogManager.getLogger(ConfigurationRepository.class);
+
+    public static class SecurityClusterStateUpdateTask extends ClusterStateUpdateTask {
+
+        private final Map<String, Integer> versions;
+
+        public SecurityClusterStateUpdateTask(final Map<String, Integer> versions) {
+            super(Priority.IMMEDIATE);
+            this.versions = versions;
+        }
+
+        @Override
+        public ClusterState execute(ClusterState currentState) throws Exception {
+            final SecurityMetadata securityMetadata = currentState.custom(SecurityMetadata.TYPE, SecurityMetadata.DEFAULT);
+            if (!securityMetadata.initialSecurityConfigurationApplied()) {
+                return ClusterState.builder(currentState).putCustom(SecurityMetadata.TYPE, new SecurityMetadata(true, versions)).build();
+            }
+            return currentState;
+        }
+
+        @Override
+        public void onFailure(String s, Exception e) {
+            LOGGER.error(s, e);
+        }
+
+    }
+
+    public static ClusterStateListener onClusterManagerListener(
+        final ClusterService clusterService,
+        final ThreadPool threadPool,
+        final ConfigurationRepository configurationRepository
+    ) {
+        return new LocalNodeClusterManagerListener() {
+
+            private Future<Void> initDefaultConfigFuture;
+
+            @Override
+            public void onClusterManager() {
+                final var securityMetadata = clusterService.state().custom(SecurityMetadata.TYPE, SecurityMetadata.DEFAULT);
+                if (!securityMetadata.initialSecurityConfigurationApplied()) {
+                    if (initDefaultConfigFuture == null) {
+                        initDefaultConfigFuture = threadPool.generic().submit(() -> {
+                            configurationRepository.initDefaultConfigurationOnManager();
+                            clusterService.submitStateUpdateTask(
+                                "init-security-configuration",
+                                new SecurityClusterStateUpdateTask(configurationRepository.versions())
+                            );
+                            return null;
+                        });
+                    }
+                }
+            }
+
+            @Override
+            public void offClusterManager() {
+                if (initDefaultConfigFuture != null) {
+                    LOGGER.warn(
+                        "Management node has been changed during configuration initialization. "
+                            + "Default configuration will be re-init from scratch on the next selected management node."
+                    );
+                    initDefaultConfigFuture.cancel(true);
+                    initDefaultConfigFuture = null;
```

Comment:
not sure that this is the right way to stop future

Reply:
I'm not clear what the intention behind initDefaultConfigFuture, can you help me understand how you'd like the flow of control to work?

Reply:
Ahhh I observed it when I re-ran integration tests 100 times. Sometimes we have problems with ports as result the manager node startes the initialization and after that switches to another node which could be the test framework problem we use. I haven't seen such problems with in the real env.  So I will remove it so far and in case there is such problem I will add such logic.